pkg/amqp/publisher.go

```diff
@@ -11,29 +12,62 @@ import (
 )
 
 type Publisher struct {
-	*connectionWrapper
+	*ConnectionWrapper
 
 	config                  Config
 	publishBindingsLock     sync.RWMutex
 	publishBindingsPrepared map[string]struct{}
+	closeConnection         func() error
 }
 
-func NewPublisher(config Config, logger watermill.LoggerAdapter) (*Publisher, error) {
+func NewPublisher(config Config, logger watermill.LoggerAdapter, opts ...Opt) (*Publisher, error) {
```

Comment:
In general, we consider "Functional Options" as an anti-pattern because it's very implicit what can be passed (it's harder to find what options are available, etc.) Instead, we are using the `Config` struct or a separate constructor. 

I think that in that case, it would be better to add a new constructor, like `NewPublisherWithConnection` for example. 
We will also avoid breaking change :)